helm/flink-kubernetes-operator/templates/flink-operator.yaml

```diff
@@ -55,6 +55,24 @@ spec:
       {{- if .Values.operatorPod.nodeSelector }}
       nodeSelector: {{ toYaml .Values.operatorPod.nodeSelector | nindent 8 }}
       {{- end }}
+      {{- if .Values.operatorPod.affinity }}
+      affinity:
+        podAntiAffinity: {{ if eq .Values.operatorPod.affinity.type "requiredDuringSchedulingIgnoredDuringExecution"}}
+          {{ .Values.operatorPod.affinity.type }}:
+          - labelSelector:
+              matchLabels:
+                {{- include "flink-operator.labels" . | nindent 16 }}
+            topologyKey: "kubernetes.io/hostname"
+          {{ else }}
+          {{ .Values.operatorPod.affinity.type }}:
+            - weight: 100
+              podAffinityTerm:
+                labelSelector:
+                  matchLabels:
+                    {{- include "flink-operator.labels" . | nindent 20 }}
+                topologyKey: "kubernetes.io/hostname"
```

Comment:
Do we need to hardcode this affinity setup?
I think it would be more robust if the user would simply specify the `affinity` object in the yaml and we use it as-is like for other more advanced features

Reply:
Thank you for bringing up the discussion. I've given it some thought, and I agree with your suggestion to let users specify the affinity object so we can enhance robustness and flexibility. I believe this change will contribute to a more adaptable system. Let me create a JIRA ticket and push new changes.